src/lib.rs

```diff
@@ -61,17 +108,104 @@ pub fn check_version(name: &str, current_version: &str) -> Result<(), Box<dyn st
     Ok(())
 }
 
+#[cfg(test)]
```

Comment:
just purely organizational point of view , let's put tests in a separate files . Example: https://github.com/hyperium/hyper/tree/master/tests

Reply:
Put it in a separate file for now rather than a directory since putting it in a "tests" directory does not set cfg(test)